src/main/preference.js

```diff
@@ -93,6 +94,32 @@ class Preference {
       }
     })
   }
+
```

Comment:
Hi, @fxha we can use [`json schema`](http://json-schema.org/) to valid user setting! but I think we can put it after version 1.0.0(or you want to do before version 1.0.0). So you can add a TODO?

webpack used [ajv](https://github.com/epoberezkin/ajv)